Dockerfile

```diff
@@ -1,93 +1,131 @@
+ARG VERSION=unknown
+RUN echo "APP_VERSION = '$VERSION'" > app/config/version.py
 
 ARG GIT_HASH=unknown
 ARG GIT_BRANCH=unknown
 ARG GIT_DIRTY=""
 ARG AUTHOR=unknown
-ARG VERSION=unknown
-ARG TARGET=
 LABEL git.hash=$GIT_HASH
 LABEL git.branch=$GIT_BRANCH
 LABEL git.dirty="$GIT_DIRTY"
 LABEL author=$AUTHOR
 LABEL version=$VERSION
 
-# Overwrite the version.py from source with the actual version
-RUN echo "APP_VERSION = '$VERSION'" > /app/config/version.py
+WORKDIR ${INSTALL_DIR}/app/
+USER ${USER}
 
 ###########################################################
-# Container to perform tests/management/dev tasks
-FROM base as debug
-
-LABEL target=debug
-
-RUN cd /tmp && \
-    pipenv install --system --deploy --ignore-pipfile --dev
-
-# this is only used with the docker-compose setup within CI
-# to ensure that the app is only started once the DB container
-# is ready
-COPY ./wait-for-it.sh /app/
-
-# for testing/management, settings.py just imports settings_dev
-RUN echo "from .settings_dev import *" > /app/config/settings.py \
-    && chown geoadmin:geoadmin /app/config/settings.py
-
-# Collect static files, uses the .env.default settings to avoid django raising settings error
-RUN APP_ENV=default LOGGING_CFG=0 ./manage.py collectstatic --noinput
+# Builder container
+FROM base as builder
+RUN apt-get update -qq \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y gdal-bin \
+    # dev dependencies
+    binutils libproj-dev \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && pip3 install pipenv \
+    && pipenv --version
 
-USER geoadmin
+ENV PIPENV_VENV_IN_PROJECT=1
+COPY Pipfile.lock Pipfile ./
+RUN pipenv sync
 
-EXPOSE $HTTP_PORT
+COPY --chown=${USER}:${GROUP} .env.default ./
+COPY --chown=${USER}:${GROUP} spec/ spec/
+COPY --chown=${USER}:${GROUP} app/ app/
 
-# entrypoint is the manage command
-ENTRYPOINT ["python3"]
+# on prod, settings.py needs to be replaced to import settings_prod instead of settings_dev
+RUN echo "from .settings_prod import *" > app/config/settings.py \
+    && chown ${USER}:${GROUP} app/config/settings.py
 
 
 ###########################################################
 # Container to use in production
 FROM base as production
-
 LABEL target=production
 
-# on prod, settings.py just import settings_prod
-RUN echo "from .settings_prod import *" > /app/config/settings.py \
-    && chown geoadmin:geoadmin /app/config/settings.py
+RUN apt-get update -qq \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y gdal-bin \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
 
-# Collect static files, uses the .env.default settings to avoid django raising settings error
-RUN APP_ENV=default LOGGING_CFG=0 ./manage.py collectstatic --noinput
+COPY --from=builder ${INSTALL_DIR} ${INSTALL_DIR}
 
-# production container must not run as root
-USER geoadmin
+# Activate virtual environnment
+ENV VIRTUAL_ENV=${INSTALL_DIR}/.venv
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
```

Comment:
To activate I would also make sure that the `PYTHONHOME` environment variable is unset, like does the `activate` script.

Reply:
Finally I have still added PYTHONHOME now, tell me if it is good like this.